tests/mainTests.cpp

```diff
@@ -191,10 +198,10 @@ void copyingTests()
     assertEquals(testList.size(), copiedList.size());
     for (int i=0; i < testList.size(); i++)
         assertEquals(testList.get(i), copiedList.get(i));
-    auto testListIter = testList.iterator();
-    auto copiedListIter = copiedList.iterator();
-    while (testListIter->hasNext())
-        assertEquals(testListIter->next(), copiedListIter->next());
+    ListIterator testListIter(testList);
+    ListIterator copiedListIter(copiedList);
+    while (testListIter.hasNext())
+        assertEquals(testListIter.next(), copiedListIter.next());
```

Comment:
After this loop add assertions that check if both iterators hasNext() methods return false